### DISCUSSION TITLE
How to send response with 304 status

### DISCUSSION THREAD

**Discussion by bestuiexperience** [vercel/next.js#15163] [Help]
I want to be able to send a response with a status code of 304 when an hTMLfile has expored from the cache. I thought Next JS by default sets ETAG headers so that this automatically happens. I am not seeing any ETAGs t or if-none-match headers being sent. How is this accomplished with Next JS?

**Comment by timneutkens** [comment]
Can you try next@canary?

**Reply by bestuiexperience** [reply]
Can you elaborate? Are you saying I need to install an additional library?

**Reply by timneutkens** [reply]
I'm saying can you try out the canary releases of Next.js to see if we've solved this already for you `npm install next@canary`

**Reply by bestuiexperience** [reply]
OK. but I don't think we can use Canary for production. So, will this fix be incorporated into the next official release of Next JS?

**Reply by timneutkens** [reply]
Yes.
